documentation/design-docs/ipc-protocol.md

```diff
@@ -0,0 +1,555 @@
+    <td colspan="2">command</td>
+    <td colspan="2">reserved</td>
+    <td colspan="8">PID</td>
+    <td colspan="4">circularBufferMB</td>
+    <td colspan="4">outputPath Length</td>
+    <td colspan="16">outputPath String</td>
+    <td colspan="4">n Providers</td>
+    <td colspan="8">Keywords</td>
+    <td colspan="4">logLevel</td>
+    <td colspan="4">provider_name length</td>
+    <td colspan="14">provider_name string</td>
+  </tr>
+  <tr>
+    <td colspan="14">"DOTNET_IPC_V1"</td>
+    <td colspan="2">66</td>
+    <td colspan="2">0x0202</td>
+    <td colspan="2">0x0000</td>
+    <td colspan="8">1234567</td>
+    <td colspan="4">250</td>
+    <td colspan="4">16</td>
+    <td colspan="16">"/tmp/foo.netperf"</td>
+    <td colspan="4">1</td>
+    <td colspan="8">100</td>
+    <td colspan="4">2</td>
+    <td colspan="4">14</td>
+    <td colspan="14">"MyEventSource"</td>
+  </tr>
+</table>
+
+Where `0x0202` is the Command to start streaming with EventPipe.
+
+### Commands
+
+Commands are a `command_set` and a `command_id`.  A `command_set` is analogous to a namespace for `command_id`s.  The typical grouping is by service running on the Diagnostic Server, e.g., there is a `command_set` for EventPipe.  This allows multiple services to have the same `command_id`s without clashing.  The combination of a `command_set` and a `command_id` encodes the Command being invoked on the Diagnostic Server.
+
+The current set of `command_set`s and `command_id`s are listed below:
+
+```c
+enum class CommandSet : uint8_t
+{
+    // reserved = 0x00,
+    Miscellaneous = 0x01,
+    EventPipe     = 0x02,
+    // future
+
+    Server = 0xFF,
+};
+```
+
+```c
+enum class ServerCommandId : uint8_t
+{
+    OK    = 0x00,
+    Error = 0xFF,
+};
+```
+
+Commands may use the generic `{ magic="DOTNET_IPC_V1"; size=20; command_set=0xFF (Server); command_id=0x00 (OK); reserved = 0x0000; }` to indicate success rather than having a command specific success `command_id`.  Similarly, Commands may use the `command_set=0xFF (Server); command_id=0xFF (Error);` to generically indicate an error has occurred.
+
+For example, the Command to start a stream session with EventPipe would be `0x0202` made up of `0x02` (the `command_set` for EventPipe) and `0x02` (the `command_id` for CollectTracing).
+
+#### EventPipe Commands
+
+```c
+enum class EventPipeCommandId : uint8_t
+{
+    // reserved = 0x00,
+    StopTracing    = 0x01, // stop a given session
+    CollectTracing = 0x02, // create/start a given session
+}
+```
+EventPipe Payloads are encoded with the following rules:
+```c
+// X, Y, Z means encode bytes for X followed by bytes for Y followed by bytes for Z
+// uint = 4 little endian bytes
+// ulong = 8 little endian bytes
+// wchar = 2 little endian bytes, UTF16 encoding
+// array<T> = uint length, length # of Ts
+// string = (array<wchar> where the last wchar must = 0) or (length = 0)
+```
+
+The `CollectTracing` Command consumes the following Payload:
```

Comment:
I think we should flesh out the commands a little further:
a) descriptive text about what the command does and what the parameters mean
b) description of any known command specific errors

I'd think about what you would want to see on MSDN for a good API doc page and put the same type of info here.